### PR TITLE
Adding nifti_clib

### PR DESCRIPTION
**Pull request by chrisgorgo**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/nifti_clib/build.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+make nifti
```
Comment (npavlovikj) [inline recipes/nifti_clib/build.sh]: Looks like `gcc` is hard-coded in the Makefile based on the error message, so please try using `make CC=${CC} nifti` instead.

File: recipes/nifti_clib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{% set name = "nifti_clib" %}
+{% set version = "2.0.0" %}
+
+package:
+    name: {{ name|lower }}
+    version: {{ version }}
+
+source:
+    url: https://github.com/NIFTI-Imaging/{{ name }}/archive/de2bb84d6219a3671ef7f5b696f17d93276da9eb.tar.gz
+
```
Comment (npavlovikj) [inline recipes/nifti_clib/meta.yaml]: Add `sha256`.

File: recipes/nifti_clib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{% set name = "nifti_clib" %}
+{% set version = "2.0.0" %}
+
+package:
+    name: {{ name|lower }}
+    version: {{ version }}
+
+source:
+    url: https://github.com/NIFTI-Imaging/{{ name }}/archive/de2bb84d6219a3671ef7f5b696f17d93276da9eb.tar.gz
```
Comment (npavlovikj) [inline recipes/nifti_clib/meta.yaml]: Please use `https://github.com/NIFTI-Imaging/nifti_clib/archive/v2.0.0.tar.gz` instead of a commit tarball.

**Comment by npavlovikj** [comment]
Please try adding:
```
export CFLAGS="$CFLAGS -I$PREFIX/include"
export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
export CPATH=${PREFIX}/include
``` 
in `build.sh` to fix the zlib errors.

**Comment by chrisgorgo** [comment]
Thanks @npavlovikj! I am struggling to get the local docker image to work hence this slow back and forth feedback loop.

I really appreciate your hints though! We'll get there!

File: recipes/nifti_clib/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Niftilib has been developed by members of the NIFTI DFWG and volunteers in the
```
Comment (npavlovikj) [inline recipes/nifti_clib/LICENSE.txt]: Looks like there is a file named LICENSE in the tarball, so you can remove this one, and have `license_file: LICENSE` instead.

File: recipes/nifti_clib/cross-linux.cmake [hunk 5]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (npavlovikj) [inline recipes/nifti_clib/cross-linux.cmake]: This file is not used anywhere, so it should be removed.

File: recipes/nifti_clib/build.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+make CFLAGS="-I$PREFIX/include -L${PREFIX}/lib" LDFLAGS=-L${PREFIX}/lib CC=${CC} all
```
Comment (npavlovikj) [inline recipes/nifti_clib/build.sh]: Can you please use this instead:
```
export CFLAGS="$CFLAGS -I$PREFIX/include"
export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
make CC=${CC} CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" all
```
?

File: recipes/nifti_clib/build.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+mv include/* ${PREFIX}/include/
```
Comment (npavlovikj) [inline recipes/nifti_clib/build.sh]: I think it is better if you use `cp` instead of `mv`, e.g.: `cp -r include $PREFIX`

File: recipes/nifti_clib/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - zlib
+  run:
+    - expat
+    - zlib
+
+test:
+    commands:
```
Comment (npavlovikj) [inline recipes/nifti_clib/meta.yaml]: You should add a test command. If there are only library/headers created, you can check if those files exist, e.g.:
```
test:
  commands:
    - test -f ${PREFIX}/include/directory/file.h
    - test -f ${PREFIX}/lib/directory/file.a
```
where `directory`, `file` are the paths to the copied headers/libraries as part of `nifti_clib`.

File: recipes/nifti_clib/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,37 @@
+{% set name = "nifti_clib" %}
+{% set version = "2.0.0" %}
+
+package:
+    name: {{ name|lower }}
```
Comment (npavlovikj) [inline recipes/nifti_clib/meta.yaml]: From aesthetic purposes, can you please fix `meta.yaml` such that each nested row is 2 spaces in, instead of 4, e.g.:
```
extra:
  recipe-maintainers:
    - leej3
```
?

File: recipes/nifti_clib/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+about:
+    home: https://github.com/NIFTI-Imaging/nifti_clib
+    license: Public Domain
+    license_file: LICENSE
+    summary: 'C library for IO with the nifti file format.'
+
+extra:
+    recipe-maintainers:
+        - leej3
```
Comment (npavlovikj) [inline recipes/nifti_clib/meta.yaml]: Please check with `leej3` that he is ok to be listed as a maintainer here. You should add yourself here as well.

File: recipes/nifti_clib/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    - expat
+    - zlib
+
+test:
+    commands:
+
+about:
+    home: https://github.com/NIFTI-Imaging/nifti_clib
+    license: Public Domain
+    license_file: LICENSE
```
Comment (npavlovikj) [inline recipes/nifti_clib/meta.yaml]: Please also add `license_family: PUBLIC-DOMAIN`.

File: recipes/nifti_clib/meta.yaml (hunk 10 above)
Comment (chrisgorgo) [reply recipes/nifti_clib/meta.yaml]: @leej3, please confirm

**Comment by leej3** [comment]
Cool. Thanks for this @chrisgorgo. I am willing to maintain this.

**Comment by npavlovikj** [comment]
@conda-forge/staged-recipes , the recipe looks good to me, so please merge/review when you can. Thanks!

File: recipes/nifti_clib/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  sha256: 62655d8457bd04ffd80cf2f07fe8c6aac2b8b00a69fbe71f399e3d9623128630
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
```
Comment (chrisburr) [inline recipes/nifti_clib/meta.yaml]: Add make here rather than relying on the system one

File: recipes/nifti_clib/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  host:
+    - zlib
+  run:
+    - expat
+    - zlib
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/nifti1.h
+    - test -f ${PREFIX}/lib/libniftiio.a
```
Comment (chrisburr) [inline recipes/nifti_clib/meta.yaml]: Does this only provide a static library? Or also a shared one?

If there is a shared one it might make sense to include run_exports

File: recipes/nifti_clib/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - zlib
+  run:
+    - expat
```
Comment (chrisburr) [inline recipes/nifti_clib/meta.yaml]: Should this also be in host? I assume it’s being used as a shared library?

Comment (leej3) [reply recipes/nifti_clib/meta.yaml]: Thanks for the review @chrisburr. We'll work through this next week.  But in answer to your questions: yes we may consider distributing a .so instead (or both). If so, run exports shall be adjusted accordingly.

**Comment by leej3** [comment]
Can we reopen this now. I don't see an option to do so. I'm assuming I don't have the appropriate permissions.

I've pushed updates to the original branch that use the recent release v3.0.1 and makes use of the cmake build system.

Also @chrisburr I updated the dependencies. expat is indeed linked against but by cifti (which should not be distributed with this package) so I have removed it.

**Comment by leej3** [comment]
Thanks. 2 additional thoughts:
+  I'm not sure if this will work with windows. I suspect it won't. I tried to specify that it was skipped in the yaml file. Perhaps I messed that up.
+ I don't think the yum_requirements.txt is needed. That may have been needed by the make build system. I'm not sure.

**Comment by leej3** [comment]
Interesting. It does seem to have passed. I won't claim credit for that!

File: recipes/nifti_clib/yum_requirements.txt [hunk 15]
```diff
@@ -0,0 +1 @@
+csh
```
Comment (jakirkham) [inline recipes/nifti_clib/yum_requirements.txt:1]: Why is this needed?

Comment (leej3) [reply recipes/nifti_clib/yum_requirements.txt]: Looks like it wasn't alright.
